### ISSUE TITLE
anonymous functions/classes at default exports.

### ISSUE THREAD

**Issue by mysticatea** [eslint/espree#180]
`FunctionDeclaration` and `ClassDeclaration` don't allow `null` in its `id` property.
- https://github.com/estree/estree/blob/master/spec.md#functiondeclaration
- https://github.com/estree/estree/blob/master/es6.md#classdeclaration

But espree parses anonymous functions/classes at default exports to `id: null`.

I'm not sure, in this case, it should use `FunctionExpression`/`ClassExpression` instead of `FunctionDeclaration`/`ClassDeclaration`.

```js
espree.parse(
  "export default function() {}",
  {ecmaFeatures: {modules: true}}
);
console.log(JSON.stringify(ast, null, 4));
```

```json
{
    "type": "Program",
    "body": [
        {
            "type": "ExportDefaultDeclaration",
            "declaration": {
                "type": "FunctionDeclaration",
                "id": null,
                "params": [],
                "body": {
                    "type": "BlockStatement",
                    "body": []
                },
                "generator": false,
                "expression": false
            }
        }
    ],
    "sourceType": "module"
}
```

**Comment by nzakas** [comment]
This is the correct behavior in both cases. I think the spec just wasn't updated. I'll file an issue there.

**Comment by mysticatea** [comment]
Thank you!
